portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,13 +85,16 @@ private int getNumberComments(HttpServletRequest request) {
     return numberComments;
   }
 
+  int idNum = 0;
```

Comment:
This will break every time your server restarts. `idNum` will get set back to zero but you'll have comments that exist in your database that already have IDs 0, 1, 2, 3...

You want this value to get initialized to N+1 where N is the highest ID that exists in your database at the time that this class gets instantiated.